web/src/compose.js

```diff
@@ -246,6 +250,10 @@ export function send_message(request = create_message_object()) {
         // We might not have updated the draft count because we assumed the
         // message would send. Ensure that the displayed count is correct.
         drafts.sync_count();
+
+        const draft = drafts.draft_model.getDraft(request.draft_id);
+        draft.is_sending_saving = false;
+        drafts.draft_model.editDraft(request.draft_id, draft);
```

Comment:
So I think we probably need some added change that `drafts.initialize` will st `is_sending_saving` to `false` for every draft as it loads them, at least if the draft isn't in the process of being sent?

Reply:
It seems like if `send_after_reload` is `true`, then we call `compose.finish`, which will set `is_sending_saving` to `true` again and then eventually set it to `false` once it completes. This doesn't happen for every reload, but specifically for a call to `reload.initiate({` when we think we need to reload the server in the middle of a message send that's timing out.

I see the concern of having drafts indefinitely stuck in sending limbo if they get interrupted by a reload, and that can definitely happen if the user just refreshes manually in the middle of a message sending. Thanks for flagging that!

I'm not sure how we'd know which drafts are "in the process of being sent by some on-startup logic" in `drafts.initialize`. I was wondering if maybe we could reset in `drafts.initialize` and then have the reload code run after that, but it seems like `drafts.initialize` is called _after_ the reload setup happens. The only other idea I have right now is to set a separate flag for sending during reload, which feels a bit bulky and messy to me, but maybe that's the best option. I'm curious if you have any ideas around how to do this?

Reply:
I don't se a reason we couldn't do the data part of `drafts.initialize` quite early. To make that work, I think we'd just have to split `drafts.initialize` such that `set_count(Object.keys(draft_model.get()).length);` was in a separate `drafts.initialize_ui()` step that runs later.

A potentially good preparatory refactor we could merge the simplify this would be to actually split the module; a new `draft_counts_ui.ts` would have `set_counts`, and that function would be passed into `drafts.initialize` like the on-narrow hooks are to avoid import cycles. But I think the little split suggested here would suffice to resolve this issue.

Reply:
Ah great, thanks. I made an `initialize_counts` function to split this out.